src/scripts/main.js

```diff
@@ -1,11 +1,36 @@
+/* eslint-disable prettier/prettier */
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  const promise = new Promise((resolve) => {
+    element.addEventListener(eventName, (e) => {
+      e.preventDefault();
+
+      // eslint-disable-next-line no-undef
+      resolve(`It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`);
+    },
+    { once: true });
+  });
+
+  return promise;
 }
```

Comment:
Bug: You are using `preventDefault` inside your event listener. This will cancel the default action of the event, which may not be what you want. For example, it will prevent form submission when the button is clicked. Unless you specifically want to stop the default event action, it's better to remove this line.